tools/ippfind.c

```diff
@@ -1150,7 +1152,7 @@ main(int  argc,				/* I - Number of command-line args */
 
   avahi_client = avahi_client_new(avahi_simple_poll_get(avahi_poll),
 			          0, client_callback, avahi_poll, &err);
-  if (!avahi_client)
+  if (!avahi_client && err)
   {
```

Comment:
NULL avahi_client means no Bonjour/DNS-SD. If Avahi doesn't return an error on an error it is their fuckup.